packages/mdl-icon-toggle/mdl-icon-toggle.scss

```diff
@@ -0,0 +1,136 @@
+  display: flex;
+  position: relative;
+  box-sizing: border-box;
+  align-items: center;
+  justify-content: center;
+  width: 48px;
+  height: 48px;
+  padding: 12px;
+  outline: none;
+  font-size: 1.5rem;
+  cursor: pointer;
+  user-select: none;
+
+  &::after {
+    position: absolute;
+    top: 0;
+    right: 0;
+    bottom: 0;
+    left: 0;
+
+    transform: scale(0, 0);
+    transition:
+      opacity 80ms $mdl-animation-fast-out-linear-in-timing-function,
+      transform 80ms $mdl-animation-fast-out-linear-in-timing-function;
+    border-radius: 50%;
+    background-color: currentColor;
+    content: "";
+    opacity: 0;
+    pointer-events: none;
+  }
+
+  &:focus::after {
+    transform: scale(1.3, 1.3);
+    transition:
+      opacity 80ms $mdl-animation-linear-out-slow-in-timing-function,
+      transform 80ms $mdl-animation-linear-out-slow-in-timing-function;
+    opacity: .12;
+  }
+
+  @include mdl-theme-dark {
+    @include mdl-theme-prop(color, text-primary-on-dark);
+
+    &:focus::after {
+      opacity: .3;
+    }
+  }
+}
+
+.mdl-icon-toggle--primary {
+  @include mdl-theme-prop(color, primary);
+}
+
+.mdl-icon-toggle--accent {
+  @include mdl-theme-prop(color, accent);
+}
+
+.mdl-icon-toggle--primary,
+.mdl-icon-toggle--accent {
+  &:focus::after {
+    opacity: .26;
+  }
+}
+
+.mdl-icon-toggle--disabled {
+  @include mdl-theme-prop(color, text-disabled-on-light);
+  pointer-events: none;
+
+  @include mdl-theme-dark(".mdl-icon-toggle", true) {
+    @include mdl-theme-prop(color, text-disabled-on-dark);
+  }
+}
+
+/** postcss-bem-linter: end */
+
+.mdl-icon-toggle.mdl-ripple-upgraded {
+  @include mdl-ripple-base;
+  // NOTE: The spec denotes specific opacity values to use for different styles of icon buttons,
+  // either "light", "dark", or "color". Because ripples are made up of two distinct elements,
+  // we need to use an opacity value such that when two elements of the same color and opacity are
+  // layered on top of one another, they create the opacity specified in the spec.
+  // This is done by using the painter's compositing algorithm:
+  //
+  // alpha[desired] = alpha[a] + alpha[b](1 - alpha[a]).
+  //
+  // Because both alpha values are the same this reduces to:
+  //
+  // alpha[desired] = alpha[a] + alpha[a](1 - alpha[a]).
+  //
+  // We simply solve this quadratic equation for the nonnegative root between [0, 1].
+  // E.g. for black 12% opacity:
+  // https://www.wolframalpha.com/input/?i=x+%2B+x(1+-+x)+%3D+.12,+x+%3C%3D+1,+x+%3E%3D+0
+  @include mdl-ripple-bg((pseudo: "::before", opacity: .062));
+  @include mdl-ripple-fg((pseudo: "::after", opacity: .062));
+
+  @include mdl-theme-dark(".mdl-icon-toggle", true) {
+    @include mdl-ripple-bg((pseudo: "::before", base-color: white, opacity: .16));
+    @include mdl-ripple-fg((pseudo: "::after", base-color: white, opacity: .16));
+  }
+
+  // For some reason, GPU layer promotion makes ripples look terrible on Chrome
+  will-change: initial;
```

Comment:
just checked and it seems okay. The pseudo-els are still promoted into their own layer so that should be sufficient for compositing.